controllers/control_plane_init.go

```diff
@@ -0,0 +1,48 @@
+package controllers
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/cloudinit"
+)
+
+type UserDataGenerator struct {
```

Comment:
What about `BootstrapDataGenerator` or `CloudInitGenerator`?

Reply:
Yes, BootstrapDataGenerator is good!